### PR TITLE
Implements new fcm api call

### PR DESCRIPTION
**Pull request by shvedchenko**

### CODE REVIEW

File: fcm.go [hunk 1]
```diff
@@ -209,37 +210,201 @@ func (this *FcmClient) Send() (*FcmResponseStatus, error) {
...
+		Success:       resp.SuccessCount,
+		Fail:          resp.FailureCount,
+		Canonical_ids: 0, // TODO Where does it come from? Is it needed?
+		Results:       *toFcmResponseResults(&resp.Responses),
+	}
 
+	return &status
+}
+
+func toFcmResponseResults(original *[]*messaging.SendResponse) *[]map[string]string {
```
Comment (tushark19) [inline fcm.go:375]: Same as above. 
Can we please move `toFcmResponseResults` and `toFcmRespStatus` to a different package

**Comment by shvedchenko** [comment]
> Overall I think we should try to separate the formatting changes (ones made to readme, instanceid.go etc) to different PRs.
> 
> Helps reduce the chaos for the reviewer and focus only on the logic updates (made in `fcm.go`)

Cmon, 8 files aint that much to be that confused. TBH Im just anxious if I read a file see a typo and leave it just as is.

**Comment by tushark19** [comment]
>Cmon, 8 files aint that much to be that confused

It's not about the number of files. 
Was just making an observation about the the experience of reviewing the code 

>TBH Im just anxious if I read a file see a typo and leave it just as is.
We can make those changes in a separate commit or as a separate PR. 

Anyway it doesn't block PR approval but was just making an observation 🤷‍♂️

File: fcm.go [hunk 2]
```diff
@@ -40,6 +43,12 @@ var (
 	fcmServerUrl = fcm_server_url
 )
 
+var firebaseApp *firebase.App
```
Comment (tushark19) [inline fcm.go:46]: Just curious when you run it locally I suppose this would be assigned a value somewhere?

File: bin/golangci-lint
Comment (tushark19) [inline bin/golangci-lint:1]: What's this file?

File: fcm.go (hunk 2 above)
Comment (shvedchenko) [reply fcm.go:46]: yep exactly

**Review by tushark19 — APPROVED** [review]

**Comment by tushark19** [comment]
LGTM. Let's test this now!

**Pull request merged**